flang/lib/Parser/openmp-parsers.cpp

```diff
@@ -473,6 +473,19 @@ TYPE_CONTEXT_PARSER("Omp LINEAR clause"_en_US,
 // OpenMPv5.2 12.5.2 detach-clause -> DETACH (event-handle)
 TYPE_PARSER(construct<OmpDetachClause>(Parser<OmpObject>{}))
 
+// InteropTypes
+TYPE_PARSER(construct<InteropType>(
+              "TARGETSYNC" >> pure(InteropType::Kind::TargetSync) ||
+              "TARGET" >> pure(InteropType::Kind::Target)))
+
+// InteropPreference
+// TYPE_PARSER(construct<InteropPreference>(name || intConstantExpr))
```

Comment:
Check out some definitions in `expr-parsers.h`. And choose whichever suites here.

Reply:
**I tried the following code:**
`TYPE_PARSER(construct<InteropPreference>(construct<InteropPreference>(defaultCharExpr) || 
    construct<InteropPreference>(intConstantExpr)))`

But it always expects **Character** expression to be present rather than allowing either **Character** Expression or **Integer Constant** Expression inside the `InteropPreference`. But similar code is used for `OmpUpdateClause` and there it expects any one of the multiple type to be present. Is there something I am missing for adding the parser support for `InteropPreference`?

Reply:
Not sure, try to debug and fix as much as you can. I will check it out tomorrow, if possible.

Reply:
I have tried making couple of modifications but still the issue persist. If you have resolved this, please let me know. Thanks!